src/emailservice/email_server.py

```diff
@@ -120,8 +121,27 @@ def SendOrderConfirmation(self, request, context):
 class DummyEmailService(BaseEmailService):
   def SendOrderConfirmation(self, request, context):
     logger.info('A request to send order confirmation email to {} has been received.'.format(request.email))
+    if self.EncodeEmail(request.email):
```

Comment:
I know that this is just a dummy function and doesn't matter too much, but it does seem a bit strange that this is checking if True, even though EncodeEmail returns true on success.

Maybe it would be more realistic to make EncodeEmail throw an exception (like `b64encode` on the string directly instead of bytes), and then catch that exception here?

Reply:
Do you mean that EncodeEmail would still try to encode the email regardless of the flag but merely do it incorrectly in this case? I think the function does look strange it's checking True but that's mostly to add a layer of "complexity" on it

Reply:
I was imagining something like this:

```
if os.getenv('ENCODE_EMAIL').lower() == "true":
  try:
    encoded = self.EncodeEmail(request.email)
    request.email = encoded
  except:
    logger.error('Err: Could not encode email')
    context.set_code(grpc.StatusCode.INTERNAL)
...
  def EncodeEmail(self, email):
      byte_rep = email.encode('ascii')
      b64_bytes = base64.b64encode(email)
      return b64_bytes.decode('ascii')
```

This should give you the same result, while looking more like a real feature with a bug. But maybe I'm misunderstanding what you're doing here

Reply:
Ah I see - no this makes sense and it does mask the feature a lot better. I made the adjustment!